### ISSUE TITLE
Fix support for --json option

### ISSUE THREAD

**Issue by weaverryan** [symfony/webpack-encore#204]
The `--json` option (e.g. `encore production --json`) is useful for visualizing your assets: https://chrisbateman.github.io/webpack-visualizer/

There are at least 2 things that Encore currently breaks:

1) We dump some output, which breaks the JSON
2) We silence a bunch of output, which the visualizer expects

**Comment by Lyrkan** [comment]
I played a bit with that, ignoring the whole `assetOutputDisplay`/`friendlyErrorPlugin` thing and removing some extra `console.log` calls when Encore is called with the `--json` (or `-j`) option.

It works well if you directly call `./node_modules/.bin/encore` but not if you use the `yarn run` wrapper :(

**Using `./node_modules/.bin/encore production --json`:**
![2017-11-21_23-43-06](https://user-images.githubusercontent.com/850046/33100766-e73ca8b0-cf15-11e7-8e11-6093deececc1.png)

**Using `yarn encore production --json` (or `yarn run encore production --json`):**
![2017-11-21_23-41-55](https://user-images.githubusercontent.com/850046/33100774-eb733ad4-cf15-11e7-8bd3-e9f8e6e88172.png)

I'm not sure how to remove these extraneous lines...

**Comment by Lyrkan** [comment]
So, I don't think that we can do much on our side about these lines **but** they can be disabled using the `--silent` CLI option (doesn't seem to be documented yet on yarn website though).

Note that it has to be placed before the script name in order to avoid Webpack throwing an `Unknown argument` error, for instance:

```
// Doesn't work
yarn run encore production --silent --json

// Works
yarn run --silent encore production --json
```

**Comment by mstrom** [comment]
When this PR can be merged?

**Comment by weaverryan** [comment]
Just released 0.17.1 with this fix :)
